src/meta/src/hummock/compaction/picker/ttl_reclaim_compaction_picker.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_pb::hummock::hummock_version::Levels;
+use risingwave_pb::hummock::InputLevel;
+
+use crate::hummock::compaction::{CompactionInput, CompactionPicker, LocalPickerStatistic};
+use crate::hummock::level_handler::LevelHandler;
+
+pub struct TtlReclaimCompactionPicker {
+    max_reclaim_file_counts: usize,
+
+    // state
+    last_select_index: usize,
+    // todo: filter table option
+}
+
+impl TtlReclaimCompactionPicker {
+    pub fn new(max_reclaim_file_counts: usize) -> Self {
+        Self {
+            max_reclaim_file_counts,
+            last_select_index: 0,
+        }
+    }
+}
+
+impl CompactionPicker for TtlReclaimCompactionPicker {
+    fn pick_compaction(
+        &mut self,
+        levels: &Levels,
+        level_handlers: &[LevelHandler],
+
+        _stats: &mut LocalPickerStatistic,
+    ) -> Option<CompactionInput> {
+        assert!(!levels.levels.is_empty());
+        let reclaimed_level = levels.levels.last().unwrap();
+        let mut select_input_ssts = vec![];
+        let level_handler = &level_handlers[reclaimed_level.level_idx as usize];
+
+        if self.last_select_index > reclaimed_level.table_infos.len() {
+            self.last_select_index = 0;
+        }
+
+        let start_indedx = self.last_select_index;
+        for sst in &reclaimed_level.table_infos[start_indedx..] {
+            self.last_select_index += 1;
+            if level_handler.is_pending_compact(&sst.id) {
+                continue;
+            }
+
+            select_input_ssts.push(sst.clone());
+            if select_input_ssts.len() == self.max_reclaim_file_counts {
```

Comment:
`max_reclaim_bytes` is better than `file_counts` because we also limit size of compaction task in level selector